### PR TITLE
Refactor funnel's compute backends

### PR DESCRIPTION
**Pull request by adamstruck**
Moved scheduler code into 'compute'.  Some backends such as GCE and Openstack still use funnel's scheduler.

CreateTask now calls compute.backend.Submit(). 

Reasoning described in the proposal that is part of #209.

### CODE REVIEW

**Comment by adamstruck** [comment]
Additional diffs beyond #203 can be seen here:

https://github.com/adamstruck/funnel/compare/split-scheduler-service...adamstruck:phase2

File: compute/basic/backend.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+// Backend represents the funnel basic basic backend.
+type Backend struct {
+	name   string
+	conf   config.Config
+	client Client
+}
+
+// Submit submits a task via gRPC call to the funnel basic basic
```
Comment (buchanae) [inline compute/basic/backend.go]: "basic basic"

File: worker/interfaces.go [hunk 2]
```diff
@@ -11,18 +11,16 @@ type Worker interface {
...
-
+type TaskReader interface {
 	Task() (*tes.Task, error)
 	State() tes.State
-	SetState(tes.State) error
 }
 
 // TaskLogger provides write access to a task's logs.
 type TaskLogger interface {
+	SetState(tes.State) error
```
Comment (buchanae) [inline worker/interfaces.go]: I would prefer `State()`

File: compute/openstack/scaler.go [hunk 3]
```diff
@@ -70,7 +70,7 @@ func (s *Backend) StartNode(w *pbs.Node) error {
 }
 
 // ShouldStartNode tells the scaler loop which nodes
-// belong to this scheduler backend, basically.
+// belong to this scheduler backend, schedulerally.
```
Comment (buchanae) [inline compute/openstack/scaler.go]: typo

File: compute/scheduler/backend.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+var log = logger.Sub("scheduler")
+
+// database represents the interface to the database used by the backend.
+type database interface {
+	QueueTask(*tes.Task) error
+}
+
+// NewBackend returns a new scheduler ComputeBackend instance.
+func NewBackend(db database, conf config.Config) (*ComputeBackend, error) {
```
Comment (buchanae) [inline compute/scheduler/backend.go]: Should be NewComputeBackend, or ComputeBackend should just be Backend

File: website/content/docs/guides/grid-engine.md [hunk 5]
```diff
@@ -7,40 +7,31 @@ menu:
...
-
-Workers will start, execute a single task, then exit immediately so that they don't
-unfairly hold slots in the SGE queue. Funnel accesses SGE by making calls
+Funnel can be configured to submit workers to [Open Grid Engine][sge] by making calls
 to `qsub`.
 
-The Funnel server process needs to run on the same machine as the SGE master.  
-Configure Funnel to use SGE by including the following config:
+The Funnel server process needs to run on the same machine as the GE master.
+Configure Funnel to use GE by including the following config:
```
Comment (buchanae) [inline website/content/docs/guides/grid-engine.md]: SGE

**Comment by adamstruck** [comment]
TODO:

- [ ] Update design graphic for website

**Pull request merged**